python-scripts/src/main/python/dataproducts/services/misc/druid_job_submitter.py

```diff
@@ -51,19 +51,21 @@ def interpolate_config(self, report_config):
         print('String interpolation for the report config completed')
         return report_config_str
 
-    def check_schedule(self, reportSchedule,report_id):
-        if reportSchedule == 'DAILY' :
+    def check_schedule(self, report_schedule, report_id, interval_slider):
+        if report_schedule == 'DAILY':
             return True
-        elif reportSchedule == 'WEEKLY' :
-            if date.today().weekday() == 0:
+        elif report_schedule == 'WEEKLY':
+            interval_slider = int(interval_slider) if interval_slider is not None else 0
+            if date.today().weekday() == interval_slider:
                 return True
-        elif reportSchedule == 'MONTHLY' :
-            if date.today().day == 1 :
+        elif report_schedule == 'MONTHLY':
+            interval_slider = int(interval_slider) + 1 if interval_slider is not None else 1
```

Comment:
interval_slider should have min and max range. min should be 1 and max should not be more than 20?